ospf6d/ospf6_asbr.c

```diff
@@ -2316,6 +2316,36 @@ void ospf6_asbr_redistribute_reset(struct ospf6 *ospf6)
 	}
 }
 
+void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
+{
+	int type;
+	struct ospf6_redist *red;
+	char *rmap_name = NULL;
+
+	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
+		if (type == ZEBRA_ROUTE_OSPF6)
+			continue;
+		red = ospf6_redist_lookup(ospf6, type, 0);
+		if (!red)
+			continue;
+
+		if (type == DEFAULT_ROUTE) {
+			ospf6_redistribute_default_set(
+				ospf6, ospf6->default_originate);
+			continue;
+		}
+		if (ROUTEMAP_NAME(red)) {
+			rmap_name = strdup(ROUTEMAP_NAME(red));
+		}
+		ospf6_asbr_redistribute_unset(ospf6, red, type);
+		if (rmap_name) {
+			ospf6_asbr_routemap_set(red, rmap_name);
+			free(rmap_name);
```

Comment:
Like said before if you replace strdup() with just strlcpy() on a buffer local to this function, you can avoid a malloc() and also this free().

Reply:
The problem with that approach is that route-map max name size is not defined.

In the following example, I put a buf size of 1024, which is not clean.

I've commited a code with XSTRDUP and XFREE, which is cleaner I think.

```
void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
{
	int type;
	struct ospf6_redist *red;
	char buf[1024];

	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
		buf[0] = '\0';
		if (type == ZEBRA_ROUTE_OSPF6)
			continue;
		red = ospf6_redist_lookup(ospf6, type, 0);
		if (!red)
			continue;

		if (type == DEFAULT_ROUTE) {
			ospf6_redistribute_default_set(
				ospf6, ospf6->default_originate);
			continue;
		}
		if (ROUTEMAP_NAME(red)) {
			strlcpy(buf, ROUTEMAP_NAME(red), sizeof(buf));
		}
		ospf6_asbr_redistribute_unset(ospf6, red, type);
		if (buf[0])
			ospf6_asbr_routemap_set(red, buf);
		ospf6_asbr_redistribute_set(type, ospf6->vrf_id);
	}
}
```

Reply:
But the CLI MUST be taking upto a certain number of characters for the route-map name. Can we not use that limit here?

Reply:
The CLI does not have limit for route-map but xpath expression containing the name must not exceed XPATH_MAXLEN

extract of `lib/routemap_cli.c`
```
DEFPY_YANG_NOSH(
	route_map, route_map_cmd,
	"route-map WORD$name <deny|permit>$action (1-65535)$sequence",
	ROUTE_MAP_CMD_STR
	ROUTE_MAP_OP_CMD_STR
	ROUTE_MAP_SEQUENCE_CMD_STR)
{
	char xpath_action[XPATH_MAXLEN + 64];
	char xpath_index[XPATH_MAXLEN + 32];
	char xpath[XPATH_MAXLEN];
	int rv;

	snprintf(xpath, sizeof(xpath),
		 "/frr-route-map:lib/route-map[name='%s']", name);
	nb_cli_enqueue_change(vty, xpath, NB_OP_CREATE, NULL);
```

Reply:
what I can do is:

```
#define RMAP_NAME_MAXLEN XPATH_MAXLEN
```

and 

```
char buf[RMAP_NAME_MAXLEN];
```

Reply:
@pushpasis Are you OK with the define ?

Reply:
XPATH_MAXLEN is length of the entire YANG Path String. Not of the route-map name. But surely it will be creater than the max-route-map-name-len.

Reply:
Done. Thank you